### DISCUSSION TITLE
Per-directory permissions  ACL

### DISCUSSION THREAD

**Discussion by idachour59** [drakkan/sftpgo#969] [Q&A] [answered]
Hi,

I have set up sftpgo (v2.3.4) with docker (and Kubernetes) I have a use case and I do not know how to implement it. I've done several search but it was unsuccessful.

For the same directory and for the same user, I want to authorize file upload but only authorize files whose pattern is: *.dat,*.zip.

I set for MyUser : 

Permission set to  **list,upload**
_Per-directory permissions_  
 set to **/srv/sftpgo/data/fic/in/MyUser**   and list,upload

 _Per-directory pattern restrictions_ 
set to **/srv/sftpgo/data/fic/in/MyUser**   and  \*.dat,\*.zip with allowed

When i am connecting with MyUser i can upload all file extension to **/srv/sftpgo/data/fic/in/MyUser**   directory.

Did you have a solution ? It will be very appreciable.

**Comment by drakkan** [comment, marked as answer]
pattern restrictions, directory permissions etc. work on virtual paths (the paths seen by the users), not filesystem paths

**Reply by idachour59** [reply]
Thanks a lot for your reply.
Is it possible to emulate virtual paths with local in order to have the patterns restrictions ?

**Reply by drakkan** [reply]
> Thanks a lot for your reply. Is it possible to emulate virtual paths with local in order to have the patterns restrictions ?

local paths vary based on the underlying OS and storage backend. Virtual paths are always UNIX paths, so you must always use virtual paths for permissions, patterns etc. in SFTPGo
